web/src/components/navbar/NavBar.test.tsx

```diff
@@ -220,50 +250,66 @@ describe("<NavBar />", () => {
     };
 
     it("displays registration button only when isAuthenticated is false", async () => {
-      useMockUserData({});
+      const userData = generateGuestUserData();
+      useMockUserData({ ...userData });
       renderDesktopNav();
       expect(screen.getByTestId("registration-button")).toBeInTheDocument();
     });
 
     it("displays log in button only when isAuthenticated is false", async () => {
-      useMockUserData({});
+      const userData = generateGuestUserData();
+      useMockUserData({ ...userData });
       renderDesktopNav();
       expect(screen.getByTestId("login-button")).toBeInTheDocument();
     });
 
     it("displays business profile button only when the menu is open", async () => {
-      useMockUserData({});
+      const userData = generateGuestUserData();
+      useMockUserData({ ...userData });
       renderDesktopNav();
-      const menuEl = screen.getByText(Config.navigationDefaults.navBarGuestText);
-      userEvent.click(menuEl);
+      const menuEl = screen.getByText(getNavBarBusinessTitle(userData));
+      fireEvent.click(menuEl);
       await waitFor(() => {
         expect(screen.getByText(Config.navigationDefaults.profileLinkText)).toBeInTheDocument();
       });
-      userEvent.click(menuEl);
+      fireEvent.click(menuEl);
       await waitFor(() => {
         expect(screen.queryByText(Config.navigationDefaults.profileLinkText)).not.toBeInTheDocument();
       });
     });
 
+    it("displays dashboard button when the menu is open", async () => {
```

Comment:
This is a little confusing because I don't see anything in the body of this test that asserts that there is a "dashboard button"

Looking at the designs, I can see that the business title is that button I think? But a test should be readable and understandable without that context.

If it's a dashboard button, should we be testing that it is clickable and that it does something (ie, route to dashboard?)

Otherwise, we could just write the test as "it displays business name title within menu" or something similar